### PR TITLE
[demo-app-ios] Corrected behaviour of Alerts in iOS demo app (Swift, Obj-C)

### PR DESCRIPTION
**Pull request by AnatolyPristensky**
Corrected behaviour of Alerts in iOS demo app (Swift, Obj-C)

~~* [ ] Has `CHANGELOG.md` been updated?~~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
~~* [ ] Did you add unit tests?~~
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Сorrected behaviour of Alerts in iOS demo app (Swift, Obj-C)

### CODE REVIEW

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 1]
```diff
@@ -320,7 +320,9 @@ extension AppDelegate: DistributeDelegate {
     NSLog("distributeNoReleaseAvailable invoked");
     let alert = UIAlertController(title: nil, message: "No updates available", preferredStyle: .alert)
     alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
-    self.window?.rootViewController?.present(alert, animated: true)
+    if MSDistributeViewController.checkPressed == true {
+    Utilities.topMostController()?.present(alert, animated: true)
```
Comment (AnastasiaKubova) [inline Sasquatch/SasquatchSwift/AppDelegate.swift:324]: formatting

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 2]
```diff
@@ -385,19 +385,21 @@ - (BOOL)distribute:(MSACDistribute *)distribute releaseAvailableWithDetails:(MSA
...
-    [self.window.rootViewController presentViewController:alertController animated:YES completion:nil];
+    [Utilities.topMostController presentViewController:alertController animated:YES completion:nil];
     return YES;
   }
   return NO;
 }
 
 - (void)distributeNoReleaseAvailable:(MSACDistribute *)distribute {
   NSLog(@"distributeNoReleaseAvailable invoked");
+  if (MSDistributeViewController.checkPressed == true){
```
Comment (AnastasiaKubova) [inline Sasquatch/SasquatchObjC/AppDelegate.m:396]: formatting

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 3]
```diff
@@ -80,7 +81,9 @@ class MSDistributeViewController: UITableViewController, AppCenterProtocol {
     switch (indexPath.section) {
     case 0:
         switch (indexPath.row) {
-        case 2: appCenter.checkForUpdate()
+        case 2:
+            MSDistributeViewController.checkPressed = true
```
Comment (AnastasiaKubova) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift:85]: formatting

File: Sasquatch/Sasquatch/Utilities.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    @objc static func topMostController() -> UIViewController? {
+        guard
+            let window = UIApplication.shared.keyWindow, var topController = window.rootViewController else {
+            return nil
+        }
+
+        while let newTopController = topController.presentedViewController {
+            topController = newTopController
+        }
+
```
Comment (AnastasiaKubova) [inline Sasquatch/Sasquatch/Utilities.swift:17]: please remove empty lines or add a comment

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 5]
```diff
@@ -4,13 +4,14 @@
 import UIKit
 import AppCenterDistribute
 
-class MSDistributeViewController: UITableViewController, AppCenterProtocol {
+@objc class MSDistributeViewController: UITableViewController, AppCenterProtocol {
```
Comment (AnastasiaKubova) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift:7]: Why we need `@objc class`?

Comment (AnatolyPristensky) [reply Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]: Sorry for the typo.

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 6]
```diff
@@ -80,7 +80,8 @@ class MSDistributeViewController: UITableViewController, AppCenterProtocol {
     switch (indexPath.section) {
     case 0:
         switch (indexPath.row) {
-        case 2: appCenter.checkForUpdate()
```
Comment (DmitriyKirakosyan) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift:83]: Let's revert this change.

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 7]
```diff
@@ -30,7 +30,17 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CrashesDelegate, CLLocati
   private var locationManager : CLLocationManager = CLLocationManager()
 
   var window: UIWindow?
-
+  func topMostViewController() -> UIViewController? {
+    guard
+      let window = UIApplication.shared.keyWindow, var topController = window.rootViewController else {
```
Comment (DmitriyKirakosyan) [inline Sasquatch/SasquatchSwift/AppDelegate.swift:35]: Please use ivar `window`, eg:
guard let topController = window?.rootViewController else {

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 8]
```diff
@@ -30,7 +30,17 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CrashesDelegate, CLLocati
...
+        return nil
+      }
+      while let newTopController = topController.presentedViewController {
+        topController = newTopController
+      }
+    return topController
```
Comment (DmitriyKirakosyan) [inline Sasquatch/SasquatchSwift/AppDelegate.swift:41]: formatting issue?

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 9]
```diff
@@ -51,6 +51,15 @@ @interface AppDelegate () <
 
 @implementation AppDelegate
 
+- (UIViewController *)topMostViewController {
+  UIWindow *keyWindow = [[[UIApplication sharedApplication] windows] firstObject];
```
Comment (DmitriyKirakosyan) [inline Sasquatch/SasquatchObjC/AppDelegate.m:55]: Please use `self.window` instead.

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 10]
```diff
@@ -30,7 +30,17 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CrashesDelegate, CLLocati
   private var locationManager : CLLocationManager = CLLocationManager()
 
   var window: UIWindow?
-
+  func topMostViewController() -> UIViewController? {
+    let window = UIApplication.shared.keyWindow
```
Comment (DmitriyKirakosyan) [inline Sasquatch/SasquatchSwift/AppDelegate.swift:34]: this variable is not needed. You already have a defined ivar `window` on line 32

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 11]
```diff
@@ -30,7 +30,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CrashesDelegate, CLLocati
...
   var window: UIWindow?
-
+  func topMostViewController() -> UIViewController? {
+    guard var topController = window?.rootViewController else {
+      return nil
+    }
+    while let newTopController = topController.presentedViewController {
+      topController = newTopController
+    }
+  return topController
```
Comment (DmitriyKirakosyan) [inline Sasquatch/SasquatchSwift/AppDelegate.swift:40]: formatting

**Review by DmitriyKirakosyan — APPROVED** [review]
